src/add-ons/kernel/drivers/nvmm/nvmm_internal.h

```diff
@@ -141,10 +153,16 @@ extern struct nvmm_owner nvmm_root_owner;
 extern volatile unsigned int nmachines;
 extern const struct nvmm_impl *nvmm_impl;
 
+#if defined(__HAIKU__) && defined(__cplusplus)
```

Comment:
Rather than add this here, it probably makes sense to just include this file in C++ files inside an `extern "C"` block.

Reply:
Yes, makes sense